libsolidity/analysis/TypeChecker.cpp

```diff
@@ -608,6 +608,20 @@ bool TypeChecker::visit(VariableDeclaration const& _variable)
 		}
 	}
 
+	auto fixedPoint = dynamic_cast<FixedPointType const*>(_variable.annotation().type);
+	if (fixedPoint)
+	{
+		bigint fixedPointTypeSize = boost::multiprecision::pow(bigint(2), fixedPoint->numBits());
+		bigint digitSize = boost::multiprecision::pow(bigint(10), fixedPoint->fractionalDigits());
```

Comment:
Does this happen before the maximum values for `numBits()` and `fractionalDigits()` are validated? If not then it should. Very high values can result in an expensive calculation.

If it does, then I think it would still be a good idea to have assertions against this here to ensure that it's always the case.

Could you also add a test for something like `ufixedMxN` where `M` and/or `N` are very large values?

Reply:
max digits is imited to 80, so no performance problme

Reply:
But it's limited by another check during analysis, right? This check here is part of analysis too so it depends on which of the two runs first.

Reply:
It's in the constructor of the type.

Reply:
oh, ok